.github/workflows/CodeCoveraReport.yaml

```
@@ -0,0 +1,77 @@
+name: Code coverage report to Codecov
+
+on:
+    push:
+      branches: 
+        - master
+jobs:
+        
+  test_client:
+    runs-on: ubuntu-latest
+        
+    name: Create and upload coverage
+    steps:
+        
+      - name: Setup Java
+        uses: actions/setup-java@v1
+        with:
+          java-version: 8
+            
+      - name: Checkout code           
+        uses: actions/checkout@v2
+        
+      - name: Install Necessary Packages
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y net-tools libssl-dev gdb
+          
+      - name: Install Boost
+        run: |
+          sudo ./scripts/install-boost.sh 1.76.0
+          
+      - name: Install Thrift
+        run: |
+          sudo ./scripts/install-thrift.sh 0.13.0
+         
+      - name: Setup Gcovr   
+        run: |
+            sudo apt install gcovr
+          
+      - name: Build
+        env:
+          BUILD_DIR: build
+          COVERAGE: ON
+          HAZELCAST_ENTERPRISE_KEY: ${{ secrets.HAZELCAST_ENTERPRISE_KEY }}
+        run: |
+          ./scripts/build-unix.sh                                          \
+              -DCMAKE_BUILD_TYPE=Debug                                     \
+              -DBUILD_SHARED_LIBS=ON                                       \
+              -DWITH_OPENSSL=ON                                            \
+              -DBUILD_TESTS=ON                                             \
+              -DBUILD_EXAMPLES=OFF
+
+      - name: Test
+        env:
+          BUILD_DIR: build
+          HAZELCAST_ENTERPRISE_KEY: ${{ secrets.HAZELCAST_ENTERPRISE_KEY }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          HZ_TEST_AWS_INSTANCE_PRIVATE_IP: ${{ secrets.HZ_TEST_AWS_INSTANCE_PRIVATE_IP }}            
+        run: |
+          ulimit -c unlimited
+          sudo sh -c "echo 'core' > /proc/sys/kernel/core_pattern"
+          sudo sh -c "echo '1' > /proc/sys/kernel/core_uses_pid"
+          ./scripts/test-unix.sh       
+       
+      - name: Collect coverage files
+        run: |   
+              gcovr --xml-pretty -o cpp_coverage.xml                \
+                --delete                                            \
+                -r "$(pwd)/hazelcast" -e "$(pwd)/hazelcast/test"    \
+                "$(pwd)/build"
```

indentation is inconsistent with other commands
```suggestion
        run: |   
          gcovr --xml-pretty -o cpp_coverage.xml                \
            --delete                                            \
            -r "$(pwd)/hazelcast" -e "$(pwd)/hazelcast/test"    \
            "$(pwd)/build"
```